packages/react-components/react-breadcrumb/src/components/BreadcrumbButton/useBreadcrumbButtonStyles.ts

```diff
@@ -0,0 +1,64 @@
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+import type { BreadcrumbButtonSlots, BreadcrumbButtonState } from './BreadcrumbButton.types';
+import type { SlotClassNames } from '@fluentui/react-utilities';
+import { useButtonStyles_unstable } from '@fluentui/react-button';
+import { tokens, typographyStyles } from '@fluentui/react-theme';
+
+export const breadcrumbButtonClassNames: SlotClassNames<BreadcrumbButtonSlots> = {
+  root: 'fui-BreadcrumbButton',
+  icon: 'fui-BreadcrumbButton__icon',
+};
+
+/**
+ * Styles for the root slot
+ */
+const useStyles = makeStyles({
+  root: {},
+  small: {
+    height: '24px',
+    ...shorthands.padding(tokens.spacingHorizontalSNudge),
+    ...typographyStyles.caption1,
+  },
+  medium: {
+    height: '32px',
+    ...shorthands.padding(tokens.spacingHorizontalSNudge),
+    ...typographyStyles.body1,
+  },
+  large: {
+    height: '40px',
+    ...shorthands.padding(tokens.spacingHorizontalS),
+    ...typographyStyles.body2,
+  },
+  currentSmall: {
+    ...typographyStyles.caption1Strong,
+  },
+  currentMedium: {
+    ...typographyStyles.body1Strong,
+  },
+  currentLarge: {
+    ...typographyStyles.subtitle2,
+  },
+});
+
+/**
+ * Apply styling to the BreadcrumbButton slots based on the state
+ */
+export const useBreadcrumbButtonStyles_unstable = (state: BreadcrumbButtonState): BreadcrumbButtonState => {
+  const styles = useStyles();
+  const currentSizeMap = {
+    small: styles.currentSmall,
+    medium: styles.currentMedium,
+    large: styles.currentLarge,
+  };
+  state.root.className = mergeClasses(
+    breadcrumbButtonClassNames.root,
+    styles[state.size],
+    styles.root,
+    state.current && currentSizeMap[state.size],
+    state.root.className,
+  );
+
+  useButtonStyles_unstable({ ...state });
```

Comment:
```suggestion
  useButtonStyles_unstable(state);
```

As we mutate `state`, I don't think that using spread here is a good idea